File: neatformcore/src/main/java/com/nerdstone/neatformcore/domain/view/RulesHandler.java
package com.nerdstone.neatformcore.domain.view;

import com.nerdstone.neatformcore.domain.model.NFormViewDetails;
import com.nerdstone.neatformcore.rules.RulesFactory;

public interface RulesHandler {

    void evaluateRule(NFormViewDetails viewDetails);

    RulesFactory getRulesFactory();

    enum Operation {
        HIDE, SHOW, DISABLE, ENABLE, FILTER
    }

}

File: neatformcore/src/main/java/com/nerdstone/neatformcore/domain/view/RulesHandler.kt
package com.nerdstone.neatformcore.domain.view

import com.nerdstone.neatformcore.domain.builders.FormBuilder
import org.jeasy.rules.api.Facts
import org.jeasy.rules.api.Rule
import org.jeasy.rules.api.Rules

interface RulesHandler {

    var executableRulesList: HashSet<Rule>

    var formBuilder: FormBuilder

    val viewIdsMap: HashMap<String, Int>

    fun handleReadOnlyState()

    fun handleFilter(filterItems: List<String>)

    fun handleSkipLogic(evaluationResult: Boolean, rule: Rule?, facts: Facts?)

    fun hideOrShowViews(facts: Facts?)

    fun hideViewsInitially(allRules: Rules?)

}
